modules/carousel/jetpack-carousel.php

```diff
@@ -542,7 +542,7 @@ class_exists( 'Jetpack_AMP_Support' )
 			$extra_data = apply_filters( 'jp_carousel_add_data_to_container', $extra_data );
 			foreach ( (array) $extra_data as $data_key => $data_values ) {
 				$html = str_replace( '<div ', '<div ' . esc_attr( $data_key ) . "='" . json_encode( $data_values ) . "' ", $html );
-				$html = str_replace( '<ul class="wp-block-gallery', '<ul ' . esc_attr( $data_key ) . "='" . json_encode( $data_values ) . "' class=\"wp-block-gallery", $html );
+				$html = str_replace( '<ul class="blocks-gallery-grid', '<ul ' . esc_attr( $data_key ) . "='" . json_encode( $data_values ) . "' class=\"blocks-gallery-grid", $html );
```

Comment:
It's most likely best if we keep supporting both formats, so Carousel works regardless of the version of WordPress / Gutenberg you used to create your gallery.

Reply:
Late Friday in a rush 🙃

Reply:
I noticed a small problem with the `str_replace()` call. Your change didn't introduce the problem, but maybe it could be fixed in this PR. This `str_replace()`will add only the first `$data_key`/`$data_value` pair in `$extra_data` to the ul element. The first loop changes the element to `<ul key=‘value’ class=“blocks-gallery-grid`. So, the string `<ul class="blocks-gallery-grid` no longer exists.